### PR TITLE
Parse wrapped sync endpoint error

### PR DESCRIPTION
**Pull request by yunfeng-scale**
# Pull Request Summary

sometimes requests are wrapped and so are error details. do one extra step to get the data out

## Test Plan and Usage Guide

with the help of @Tma2333-Scale, tested for exception within the API (verified that we can get traceback),

and for exception from Pydantic before reaching to the API:
image

### CODE REVIEW

File: model-engine/model_engine_server/infra/gateways/live_sync_model_endpoint_inference_gateway.py [hunk 1]
```diff
@@ -186,12 +186,22 @@ async def predict(
...
-                    if isinstance(error_json, dict)
-                    else None
-                )
+                if not isinstance(error_json, dict):
+                    result_traceback = None
+                else:
+                    if "result" in error_json:
+                        error_json = error_json["result"]
+                        if not isinstance(error_json, dict):
+                            error_json = orjson.loads(error_json)
```
Comment (yixu34) [inline model-engine/model_engine_server/infra/gateways/live_sync_model_endpoint_inference_gateway.py:197]: Can this fail?

File: model-engine/model_engine_server/infra/gateways/live_sync_model_endpoint_inference_gateway.py [hunk 2]
```diff
@@ -186,12 +186,22 @@ async def predict(
...
             logger.error(f"Service error on sync task: {exc.content!r}")
             try:
+                # Try to parse traceback from the response, fallback to just return all the content if failed
                 error_json = orjson.loads(exc.content.decode("utf-8"))
-                result_traceback = (
-                    error_json.get("detail", {}).get("traceback")
-                    if isinstance(error_json, dict)
-                    else None
-                )
+                if not isinstance(error_json, dict):
```
Comment (yixu34) [inline model-engine/model_engine_server/infra/gateways/live_sync_model_endpoint_inference_gateway.py:191]: Wonder if we can reduce nesting

File: model-engine/model_engine_server/infra/gateways/live_sync_model_endpoint_inference_gateway.py (hunk 1 above)
Comment (yunfeng-scale) [reply model-engine/model_engine_server/infra/gateways/live_sync_model_endpoint_inference_gateway.py:197]: if this fails it'll go to L209 and we'll return everything unpacked

File: model-engine/model_engine_server/infra/gateways/live_sync_model_endpoint_inference_gateway.py (hunk 2 above)
Comment (yunfeng-scale) [reply model-engine/model_engine_server/infra/gateways/live_sync_model_endpoint_inference_gateway.py:191]: there are 3 cases i'm checking:
detail.traceback
result.detail.traceback
result.detail[]

i could do simplification i guess

**Review by yixu34 — APPROVED** [review]

**Pull request merged**